pinot-common/src/main/java/org/apache/pinot/common/http/MultiGetRequest.java

```diff
@@ -94,7 +95,7 @@ public MultiGetRequest(Executor executor, HttpConnectionManager connectionManage
    * @return instance of CompletionService. Completion service will provide
    *   results as they arrive. The order is NOT same as the order of URLs
    */
-  public CompletionService<GetMethod> execute(List<String> urls, int timeoutMs) {
+  public CompletionService<GetMethod> execute(List<String> urls, Map<String, String> requestHeaders, int timeoutMs) {
```

Comment:
```suggestion
  public CompletionService<GetMethod> execute(List<String> urls, @Nullable Map<String, String> requestHeaders, int timeoutMs) {
```

Reply:
Let's also keep the original method (without requestHeaders) so that we don't need to change all the existing usages

Reply:
Ack